tools/isobuild/import-scanner.js

```diff
@@ -0,0 +1,431 @@
+import assert from "assert";
+import {isString, has, keys, each, without} from "underscore";
+import {sha1} from "../fs/watch.js";
+import {matches as archMatches} from "../utils/archinfo.js";
+import {findImportedModuleIdentifiers} from "./js-analyze.js";
+import {
+  pathJoin,
+  pathRelative,
+  pathNormalize,
+  pathDirname,
+  pathBasename,
+  pathExtname,
+  statOrNull,
+  readFile,
+  convertToPosixPath,
+} from "../fs/files.js";
+
+export default class ImportScanner {
+  constructor({
+    name,
+    bundleArch,
+    sourceRoot,
+    extensions = [".js", ".json"],
+    usedPackageNames = {},
+    nodeModulesPath,
+  }) {
+    assert.ok(isString(sourceRoot));
+
+    this.name = name;
+    this.bundleArch = bundleArch;
+    this.sourceRoot = sourceRoot;
+    this.usedPackageNames = usedPackageNames;
+    this.nodeModulesPath = nodeModulesPath;
+    this.absPathToOutputIndex = {};
+    this.outputFiles = [];
+    this.extensions = extensions;
+  }
+
+  addInputFiles(...args) {
+    const Ap = Array.prototype;
+    const flatFiles = Ap.concat.apply(Ap, args);
```

Comment:
This code was originally written to do the work of [this code](https://github.com/meteor/meteor/blob/ff64b73253694858ecbc0446b9d88e22cfcbe35d/tools/isobuild/compiler-plugin.js#L523-L528), but I never ended up changing `getResources`.

It may be worth noting that `Array.prototype.concat.apply(Array.prototype, array)` will flatten any nested arrays embedded in the given `array` (like the code in `getResources`), so it's definitely different from both `_.toArray` and `Array.from`.